### PR TITLE
Issue #11890: Add a new action container in the EditToolbar

### LINKED ISSUE #11890: Add a new action container in the EditToolbar
We would like to add a new Action Container in between the icon and InlineAutocompleteEditText in the EditToolbar for the new unified search selector toolbar button. I think this is more reasonable compare to adding a Fenix specific toolbar button into the EditToolbar, and we can allow Apps to define what actions should be added in it similar to the DisplayToolbar with Navigation Actions.

### PR DESCRIPTION
**Pull request by gabrielluong**
Fixes #11890
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/main/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/edit/EditToolbar.kt [hunk 1]
```diff
@@ -40,13 +40,14 @@ private const val AUTOCOMPLETE_QUERY_THREADS = 3
  * Sub-component of the browser toolbar responsible for allowing the user to edit the URL ("edit mode").
  *
  * Structure:
- * +------+---------------------------+---------+------+
- * | icon |           url             | actions | exit |
- * +------+---------------------------+---------+------+
+ * +------+----------------+---------------------------+--------------+------+
+ * | icon | search actions |           url             | edit actions | exit |
```
[the remark this answers is not in the document]
Comment (gabrielluong) [reply components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/edit/EditToolbar.kt:44]: I suppose you aren't saying we should define indicators like DisplayToolbar, but more about the naming of things. I would propose that we can do something very similar to the Android naming conventions and say `addEditActionLeft` and `addEditActionRight`.

**Review by rocketsroger — APPROVED** [review]
🚢

File: components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/edit/EditToolbar.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (jonalmeida) [reply components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/edit/EditToolbar.kt]: Copying Slack messages here:

I haven't had a chance to look into what they are called there. I'm guessing it'll be something like `addDisplayIndicators` for the current display one and `addEditIndicators` for the edit mode.

[the remark this answers is not in the document]
Comment (jonalmeida) [reply components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/edit/EditToolbar.kt]: I'm still not convinced if we need to introduce a new concept of directional (left/right) edit actions to the toolbar. Android naming conventions did not consider RTL languages so have switched from left/right namings to start/end - would that be an acceptable compromise?

[the remark this answers is not in the document]
Comment (gabrielluong) [reply components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/edit/EditToolbar.kt]: Resolved with using addEditActionStart/addEditActionEnd

**Review by jonalmeida — APPROVED** [review]
LGTM, but let's wait for Roger's review as well.

**Review by rocketsroger — APPROVED** [review]

**Pull request merged**